deps/rabbitmq_prometheus/src/rabbit_prometheus_app.erl

```diff
@@ -33,52 +33,32 @@ init(_) ->
 
 -spec start_configured_listener() -> ok.
 start_configured_listener() ->
-    Listeners0 = case {has_configured_tcp_listener(),
-                       has_configured_tls_listener()} of
-                     {false, false} ->
-                         %% nothing is configured
-                         [get_tcp_listener()];
-                     {false, true} ->
-                         [get_tls_listener()];
-                     {true, false} ->
-                         [get_tcp_listener()];
-                     {true, true} ->
-                         [get_tcp_listener(),
-                          get_tls_listener()]
-                 end,
-    Listeners1 = maybe_disable_sendfile(Listeners0),
-    [start_listener(Listener) || Listener <- Listeners1],
+    case get_env(tcp_config, []) of
```

Comment:
Note that in the original version of this function there is a "nothing is configured" clause. It starts a TCP listener with all defaults.

In this implementation, nothing is started. It can be verified with

``` shell
bazel run broker RABBITMQ_ENABLED_PLUGINS="rabbitmq_prometheus"
```

Then hit Enter a couple of times to enter node shell.

Then inspect the listeners 

``` erl
rabbit_networking:active_listeners().
```

or use CLI tools:

``` shell
rabbitmq-diagnostics listeners
```

Reply:
Good catch!

Reply:
Yes this is a change in behavior. 

I looked into how other plugins implement `*.tcp.listener = none` and this is consistent with `web_mqtt` and `web_stomp`. You could turn on `web_mqtt` and `web_stomp` and set `web_mqtt.tcp.listener = none web_stomp.tcp.listener = none` which results in no active listener for these two plugins:
```
Asking node rabbit@test-server-0.test-nodes.rabbitmq-system to report its protocol listeners ...
Interface: [::], port: 15672, protocol: http, purpose: HTTP API
Interface: [::], port: 1883, protocol: mqtt, purpose: MQTT
Interface: [::], port: 61613, protocol: stomp, purpose: STOMP
Interface: [::], port: 15692, protocol: http/prometheus, purpose: Prometheus exporter API over HTTP\
...
rabbitmq@test-server-0:/$ rabbitmq-plugins list
Listing plugins with pattern ".*" ...
 Configured: E = explicitly enabled; e = implicitly enabled
 | Status: * = running on rabbit@test-server-0.test-nodes.rabbitmq-system
 |/
...
[E*] rabbitmq_web_mqtt                 3.11.10
[  ] rabbitmq_web_mqtt_examples        3.11.10
[E*] rabbitmq_web_stomp                3.11.10
[  ] rabbitmq_web_stomp_examples       3.11.10
```

I am not saying behavior of web stomp or web mqtt is a guideline for the prometheus plugin, just providing my thought process. The plugin is built with [default value for `tls_config`](https://github.com/rabbitmq/rabbitmq-server/blob/prometheus-tcp-listener-none/deps/rabbitmq_prometheus/BUILD.bazel#L26), so by default it should start with a tcp listener unless user deliberately set `prometheus.tcp.listener = none`.

What do you think?

Reply:
Alternatively we can use `[{port, 15692}]` as default in the Cuttlefish translation function.

Reply:
There is a default but for some reason it doesn't have the effect. No listeners are started if the plugin is not configured at all.

I will try to investigate.

Reply:
So it is a Cuttlefish behavior where it does not translate a proplist property the way we expect. I think it'd be easier to handle it in the code than cast some Cuttlefish magic no one would remember in a couple of months.

Reply:
Do you think we should update web_stomp and web_mqtt? They have the same issue as you've described when there is no TLS and TCP configuration, the plugin will start without listener:
https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbitmq_web_stomp/src/rabbit_web_stomp_listener.erl#L44-L52
https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbitmq_web_mqtt/src/rabbit_web_mqtt_app.erl#L83-L91